### PR TITLE
Users/kcheekuri/aciconatinerpooling

### PR DESCRIPTION
**Pull request by kcheekuri**
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/1008?feature.notebooksTemporarilyDown=false&feature.phoenix=true)

image

Features:
image
Bug Fix:
image

### CODE REVIEW

File: src/Contracts/DataModels.ts [hunk 1]
```diff
@@ -496,3 +498,8 @@ export interface MemoryUsageInfo {
   freeKB: number;
   totalKB: number;
 }
+
+export interface ContainerConnectionInfo {
+  status: ConnectionStatusType;
+  //need to add ram and rom info
```
Comment (srinathnarayanan) [inline src/Contracts/DataModels.ts:504]: we already get this from the memory call right?

File: src/Explorer/Explorer.tsx [hunk 2]
```diff
@@ -343,28 +346,51 @@ export default class Explorer {
...
+        resourceGroup: userContext.resourceGroup,
+        subscriptionId: userContext.subscriptionId,
+      };
+      const connectionInfo = await this.phoenixClient.containerConnectionInfo(provisionData);
+      if (connectionInfo.data && connectionInfo.data.notebookServerUrl) {
+        connectionStatus.status = ConnectionStatusType.Connected;
+        useNotebook.getState().setConnectionInfo(connectionStatus);
+
+        useNotebook.getState().setNotebookServerInfo({
+          notebookServerEndpoint: userContext.features.notebookServerUrl || connectionInfo.data.notebookServerUrl,
```
Comment (srinathnarayanan) [inline src/Explorer/Explorer.tsx:368]: just return the data part directly from containerConnectionInfo

File: src/Explorer/Menus/CommandBar/ConnectionStatusComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import { ProgressIndicator, Spinner, SpinnerSize, Stack, TooltipHost } from "@fluentui/react";
+import * as React from "react";
+import { ConnectionStatusType } from "../../../Common/Constants";
+import { useNotebook } from "../../Notebook/useNotebook";
+import "../CommandBar/ConnectionStatusComponent.less";
+
+export const ConnectionStatus: React.FC = (): JSX.Element => {
```
Comment (srinathnarayanan) [inline src/Explorer/Menus/CommandBar/ConnectionStatusComponent.tsx:7]: Make this a React class, look at MongoIndexingPolicyComponent for reference maybe. You ll have states and props, changing a state will change UI directly

File: src/Explorer/Tabs/NotebookV2Tab.ts [hunk 4]
```diff
@@ -370,12 +369,6 @@ export default class NotebookTabV2 extends NotebookTabBase {
   private async configureServiceEndpoints(kernelName: string) {
     const notebookConnectionInfo = useNotebook.getState().notebookServerInfo;
     const sparkClusterConnectionInfo = useNotebook.getState().sparkClusterConnectionInfo;
-    await NotebookConfigurationUtils.configureServiceEndpoints(
```
Comment (srinathnarayanan) [inline src/Explorer/Tabs/NotebookV2Tab.ts:373]: why was this removed

File: src/Contracts/DataModels.ts (hunk 1 above)
Comment (kcheekuri) [reply src/Contracts/DataModels.ts:504]: This is the initial allocation status check, will use existing memory call after connecting to network.

File: src/Explorer/Tree/ResourceTree.tsx [hunk 5]
```diff
@@ -190,32 +193,48 @@ export const ResourceTree: React.FC<ResourceTreeProps> = ({ container }: Resourc
...
-        label: "Disconnect from GitHub",
-        onClick: () => {
-          TelemetryProcessor.trace(Action.NotebooksGitHubDisconnect, ActionModifiers.Mark, {
-            dataExplorerArea: Areas.Notebook,
-          });
-          container.notebookManager?.gitHubOAuthService.logout();
-        },
-      },
-    ];
+    gitHubNotebooksTree.contextMenu = isConnected
```
Comment (srinathnarayanan) [inline src/Explorer/Tree/ResourceTree.tsx:196]: create 2 separate arrays, change to

    gitHubNotebooksTree.contextMenu = isConnected ? array1 ? array2
more readable

File: src/Phoenix/PhoenixClient.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    return {
+      status: response.status,
+      data,
+    };
+  }
+
+  public static getPhoenixEndpoint(): string {
+    const junoEndpoint = userContext.features.junoEndpoint ?? configContext.JUNO_ENDPOINT;
+    if (configContext.allowedJunoOrigins.indexOf(new URL(junoEndpoint).origin) === -1) {
+      const error = `${junoEndpoint} not allowed as juno endpoint`;
```
Comment (srinathnarayanan) [inline src/Phoenix/PhoenixClient.ts:55]: make junoclient-> phoenixclient here and everywhere else

File: src/Phoenix/PhoenixClient.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      console.error(error);
+      throw new Error(error);
+    }
+
+    return junoEndpoint;
```
Comment (srinathnarayanan) [inline src/Phoenix/PhoenixClient.ts:60]: phoenixendpoint

File: src/Utils/GalleryUtils.test.ts [hunk 8]
```diff
@@ -34,7 +34,7 @@ describe("GalleryUtils", () => {
 
     expect(useDialog.getState().visible).toBe(true);
     expect(useDialog.getState().dialogProps).toBeDefined();
-    expect(useDialog.getState().dialogProps.title).toBe("Download to My Notebooks");
+    expect(useDialog.getState().dialogProps.title).toBe("Download to My Notebooks ");
```
Comment (srinathnarayanan) [inline src/Utils/GalleryUtils.test.ts:37]: extra space

File: src/Utils/GalleryUtils.ts [hunk 9]
```diff
@@ -223,11 +224,15 @@ export function downloadItem(
 
   const name = data.name;
   useDialog.getState().showOkCancelModalDialog(
-    "Download to My Notebooks",
-    `Download ${name} from gallery as a copy to your notebooks to run and/or edit the notebook.`,
+    `Download to My Notebooks ${userContext.features.phoenix ? "Scratch" : ""}`,
```
Comment (srinathnarayanan) [inline src/Utils/GalleryUtils.ts:227]: this is used repeatedly, have a helper to choose between "My Notebooks" and "My Notebooks Scratch"

File: src/Utils/GalleryUtils.test.ts (hunk 8 above)
Comment (kcheekuri) [reply src/Utils/GalleryUtils.test.ts:37]: Had to do this to solve Lint errors at the time of PR raise.

File: src/Explorer/Explorer.tsx [hunk 10]
```diff
@@ -343,28 +346,51 @@ export default class Explorer {
...
+      });
+    }
 
     useNotebook.getState().initializeNotebooksTree(this.notebookManager);
 
     this.refreshNotebookList();
 
     this._isInitializingNotebooks = false;
   }
-
```
Comment (srinathnarayanan) [inline src/Explorer/Explorer.tsx:366]: line removed. Leave a line between functions

File: src/Phoenix/PhoenixClient.ts [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+export interface IProvosionData {
+  cosmosEndpoint: string;
+  resourceId: string;
+  dbAccountName: string;
+  aadToken: string;
+  resourceGroup: string;
+  subscriptionId: string;
+}
+export class PhoenixClient {
+  /* Retrieves the connection info of the Container */
```
Comment (srinathnarayanan) [inline src/Phoenix/PhoenixClient.ts:27]: leave comments only when doing something out of pattern. can remove this

File: src/Explorer/Explorer.tsx (hunk 2 above)
Comment (kcheekuri) [reply src/Explorer/Explorer.tsx:368]: Want to maintain consistency through out return types in phoenix client.... also having status in return call helps. May not be here but through out app level this helps.

File: src/Explorer/Tabs/NotebookV2Tab.ts (hunk 4 above)
Comment (kcheekuri) [reply src/Explorer/Tabs/NotebookV2Tab.ts:373]: Adding. Thanks

File: src/Explorer/Tree/ResourceTree.tsx (hunk 5 above)
Comment (kcheekuri) [reply src/Explorer/Tree/ResourceTree.tsx:196]: Addressed

File: src/Explorer/Menus/CommandBar/ConnectionStatusComponent.tsx (hunk 3 above)
Comment (kcheekuri) [reply src/Explorer/Menus/CommandBar/ConnectionStatusComponent.tsx:7]: With new version of react we got this latest way [Doing the same].

File: src/Phoenix/PhoenixClient.ts (hunk 11 above)
Comment (kcheekuri) [reply src/Phoenix/PhoenixClient.ts:27]: Addressed

File: src/Utils/GalleryUtils.ts (hunk 9 above)
Comment (kcheekuri) [reply src/Utils/GalleryUtils.ts]: addressed

File: src/Phoenix/PhoenixClient.ts (hunk 7 above)
Comment (kcheekuri) [reply src/Phoenix/PhoenixClient.ts]: addressed

File: src/Contracts/DataModels.ts (hunk 1 above)
Comment (srinathnarayanan) [reply src/Contracts/DataModels.ts:504]: we can add it in the next iteration

File: src/Explorer/Tree/ResourceTreeAdapter.tsx [hunk 12]
```diff
@@ -130,9 +131,12 @@ export class ResourceTreeAdapter implements ReactAdapter {
       path: "Gallery",
       type: NotebookContentItemType.File,
     };
-
+    let notebookText = ResourceTreeAdapter.MyNotebooksTitle;
```
Comment (srinathnarayanan) [inline src/Explorer/Tree/ResourceTreeAdapter.tsx:134]: move this to a util? i can still see this logic repeated in a few places. have a phoenix util, we can put more util methods there later

File: src/Explorer/Notebook/useNotebook.ts [hunk 13]
```diff
@@ -65,6 +68,8 @@ export const useNotebook: UseStore<NotebookState> = create((set, get) => ({
   myNotebooksContentRoot: undefined,
   gitHubNotebooksContentRoot: undefined,
   galleryContentRoot: undefined,
+  connectionInfo: undefined,
+  NotebookScratchText: userContext.features.phoenix ? "My Notebooks Scratch" : "My Notebooks",
```
Comment (srinathnarayanan) [inline src/Explorer/Notebook/useNotebook.ts:72]: NotebookFolderName

File: src/Explorer/Tree/ResourceTreeAdapter.tsx (hunk 12 above)
Comment (srinathnarayanan) [reply src/Explorer/Tree/ResourceTreeAdapter.tsx:134]: use the useNotebooks.get().NotebooksFolderName

**Review by srinathnarayanan — APPROVED** [review]

**Review by ramrajprabu — APPROVED** [review]
